include/boost/math/tools/simple_continued_fraction.hpp

```diff
@@ -78,29 +85,31 @@ class simple_continued_fraction {
        // The shorter representation is considered the canonical representation,
        // so if we compute a non-canonical representation, change it to canonical:
        if (b_.size() > 2 && b_.back() == 1) {
-          b_[b_.size() - 2] += 1;
-          b_.resize(b_.size() - 1);
+           b_.pop_back();
+           b_.back() += 1;
        }
        b_.shrink_to_fit();
-       
-       for (size_t i = 1; i < b_.size(); ++i) {
+
+       const int size_ = b_.size();
+       for (int i = 1; i < size_; ++i) {
```

Comment:
This will result in a warning for `-Wsign-conversion`.

Reply:
It won't, I changed it in such a way that both variables are `int`.

Reply:
`std::vector<T>::size_type` is an unsigned integer so you would need to cast to `const int size_`. Signed overflow is UB is while unsigned overflow is defined. I think this has been the case since C99.